src/xin_memory_history_manager.cpp

```diff
@@ -42,8 +42,15 @@ namespace xeus
         history_type history;
 
         int count = std::min(n, static_cast<int>(m_history.size()));
-        std::copy_n(m_history.cbegin(), count, std::back_inserter(history));
 
+        if (n > count) 
+            history = m_history;
+        else
+            std::copy(
+                m_history.rbegin(), 
+                std::next(m_history.rbegin(), n),
+                std::front_inserter(history));
```

Comment:
why not simply:
```cpp
            std::copy(
                m_history.rbegin(), 
                std::next(m_history.rbegin(), count),
                std::front_inserter(history));
```
?

This way you can remove the branches and have a single lines that works for both cases.

Reply:
ok nevermind, let's fix that when refactoring.